x-pack/plugins/inference/server/tasks/nl_to_esql/utils/validate_query_ast.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { validateQuery } from '@kbn/esql-validation-autocomplete';
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
+import { splitIntoCommands } from '../../../../common/tasks/nl_to_esql/correct_common_esql_mistakes';
+
+interface QuerySyntaxError {
+  message: string;
+  startPos: number;
+  endPos: number;
+  code?: string;
+}
+
+/**
+ * Run the query through the kibana
+ * @param query
+ */
+export const validateQueryAst = async (query: string): Promise<QuerySyntaxError[]> => {
```

Comment:
Perhaps out of scope or premature, but this feels like something that should exist within an es|ql package.

Reply:
Yeah, I want to take a closer look at the ESQL task later, as I'd like to even extract it from the `inference` plugin (this is a task, not part of the inference plugin/framework), but I was planning on doing it on a distinct PR as it's still not totally clear to me how exactly we should restructure.

(but fwiw, not sure it would benefit from living into an esql package rather than with the task esql, the function already mostly depend on low-level ESQL directives)